repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
+    depends_on("mpi", when="+mpi")
+    depends_on("cuda@11.0:", when="+cuda")
+    depends_on("hip@4.0:", when="+hip")
+    depends_on("rocm-openmp-extras", when="+hip")
+    depends_on("intel-oneapi-dpl", when="+sycl")
+    depends_on("intel-oneapi-mkl", when="+sycl")
+    depends_on("amrex@23.05:", when="@25.04")
+    depends_on("amrex@23.08:", when="@25.08")
+    depends_on("pkgconf", type="build")
+
+    with when("+netcdf"):
+        depends_on("netcdf-c+mpi+parallel-netcdf", when="+mpi")
+        depends_on("netcdf-c~mpi+parallel-netcdf", when="~mpi")
+        depends_on("netcdf-fortran")
+        depends_on("hdf5+mpi", when="+mpi")
+        depends_on("hdf5~mpi", when="~mpi")
+
+    depends_on("fftw", when="+fft")
+
+    conflicts("+cuda", when="+hip", msg="Cannot enable both CUDA and HIP")
+    conflicts("+openmp", when="+cuda", msg="Cannot enable both OpenMP and CUDA")
+    conflicts("+openmp", when="+hip", msg="Cannot enable both OpenMP and HIP")
+    conflicts("+cuda", when="+sycl", msg="Cannot enable both CUDA and SYCL")
+    conflicts("+hip", when="+sycl", msg="Cannot enable both HIP and SYCL")
+    conflicts("+fft", when="~mpi", msg="FFT support requires MPI")
+    conflicts("+radiation", when="platform=darwin", msg="Radiation is not supported on macOS")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ERF_ENABLE_MPI", "mpi"),
+            self.define_from_variant("ERF_ENABLE_OMP", "openmp"),
+            self.define_from_variant("ERF_ENABLE_NETCDF", "netcdf"),
+            self.define_from_variant("ERF_ENABLE_PARTICLES", "particles"),
+            self.define_from_variant("ERF_ENABLE_MULTIBLOCK", "multiblock"),
+            self.define_from_variant("ERF_ENABLE_RADIATION", "radiation"),
+            self.define_from_variant("ERF_BUILD_TESTS", "tests"),
+            self.define_from_variant("ERF_BUILD_FCOMPARE", "fcompare"),
+            self.define_from_variant("ERF_ENABLE_FFT", "fft"),
+            self.define("ERF_DIM", "3"),
+        ]
+        args.append(self.define("ERF_USE_EXTERNAL_AMREX", False))
+        args.append(self.define("ERF_CLONE_AMREX", True))
+        args.append(self.define("GIT_SUBMODULE_PROTOCOL", "https"))
+        args.append(self.define("MPIEXEC_PREFLAGS", "--oversubscribe"))
+        args.append(self.define("CMAKE_BUILD_TYPE", "Release"))
+        args.append(self.define("ERF_DIM", "3"))
+
+        if "+netcdf" in self.spec:
+            args.extend(
+                [
+                    self.define("NetCDF_C_PATH", self.spec["netcdf-c"].prefix),
+                    self.define("NetCDF_FORTRAN_PATH", self.spec["netcdf-fortran"].prefix),
+                ]
+            )
+
+        if "+radiation" in self.spec:
+            if "+cuda" in self.spec:
+                args.append(self.define("Kokkos_ARCH_AMPERE80", True))
```

Comment:
This should not be hardcoded to `ampere80`, it should probably take its value from the `cuda_arch` variant.

If `ampere` is the only valid cuda target for the package, please override the `cuda_arch` variant to ensure it's the only valid value. We shouldn't let people try to build for a different arch and "secretly" give them an ampere build without errors.